rs/src/management/http_client.rs

```diff
@@ -518,6 +499,33 @@ impl From<TunnelClientBuilder> for TunnelManagementClient {
     }
 }
 
+fn add_query(mut url: Url, tunnel_opts: &TunnelRequestOptions) -> Url
+{
+    if tunnel_opts.include_ports {
+        url.query_pairs_mut().append_pair("includePorts", "true");
+    }
+    if tunnel_opts.include_access_control {
+        url.query_pairs_mut().append_pair("includeAccessControl", "true");
+    }
+    if !tunnel_opts.token_scopes.is_empty() {
+        url.query_pairs_mut().append_pair("tokenScopes", &tunnel_opts.token_scopes.join(","));
+    }
+    if tunnel_opts.force_rename {
+        url.query_pairs_mut().append_pair("forceRename", "true");
+    }
+    if !tunnel_opts.tags.is_empty() {
+        url.query_pairs_mut().append_pair("tags", &tunnel_opts.tags.join(","));
+        if tunnel_opts.require_all_tags {
+            url.query_pairs_mut().append_pair("allTags", "true");
+        }
+    }
+    if tunnel_opts.limit > 0 {
+        url.query_pairs_mut().append_pair("limit", &tunnel_opts.limit.to_string());
```

Comment:
If this function took a `&mut Url`, you could keep the query pairs reference around as the original code did without having to return the Url. Not a big deal just less verbosity.

Reply:
Thanks for the tip - this is my first rust change and the feedback is appreciated. Done!